Adapters/Vungle/ISVungleAdapter/ISVungleAdapter.m

```diff
@@ -44,35 +45,11 @@ typedef NS_ENUM(NSInteger, InitState) {
 static InitState _initState = INIT_STATE_NONE;
 static ConcurrentMutableSet<ISNetworkInitCallbackProtocol> *initCallbackDelegates = nil;
 
-// Banner state possible values
-typedef NS_ENUM(NSUInteger, BANNER_STATE) {
-    UNKNOWN,
-    REQUESTING,
-    REQUESTING_RELOAD,
-    SHOWING
-};
-
-@interface ISVungleAdapter() < VungleDelegate, ISNetworkInitCallbackProtocol, InitiatorDelegate> {
-}
+@interface ISVungleAdapter() <ISNetworkInitCallbackProtocol>
 
-// Rewarded video
-@property (nonatomic, strong) ConcurrentMutableDictionary *rewardedVideoPlacementIdToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableDictionary *rewardedVideoPlacementIdToServerData;
-@property (nonatomic, strong) ConcurrentMutableDictionary *rewardedVideoServerDataToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableSet        *rewardedVideoPlacementIdsForInitCallbacks;
-
-// Interstitial
-@property (nonatomic, strong) ConcurrentMutableDictionary *interstitialPlacementIdToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableDictionary *interstitialPlacementIdToServerData;
-@property (nonatomic, strong) ConcurrentMutableDictionary *interstitialServerDataToSmashDelegate;
-
-// Banner
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToSize;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToViewController;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToBannerState;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToServerData;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerServerDataToSmashDelegate;
+@property (nonatomic, strong) ISVungleRewardedVideoAdapterRouter *rewardedVideoAdapterRouter;
+@property (nonatomic, strong) ISVungleInterstitialAdapterRouter *interstitialAdapterRouter;
```

Comment:
Where does the adapter release these references when the ad closes?

Reply:
@stanley-vungle Today I released the reference when Ad closes but found for InitInterstitial, there is only one `ISVungleAdapter` instance for it, every time when loading a new InitInterstitial Ad, the IronSource mediation will use this `ISVungleAdapter` instance and call its load Interstitial function, the init Interstitial function will be called on the first time. The mediation will not create a new `ISVungleAdapter` instance.  So we cannot release the reference when Ad closes for InitInterstitial, if we do, we cannot load a new InitInterstitial Ad.

But for Rewarded Video, the mediation will create a new `ISVungleAdapter` instance when loading a new Ad. So I removed the parentAdapter property in each ad type AdapterRouter. 
Thanks a lot. :)
cc: @ManojBudumuru-vungle

Reply:
@ManojBudumuru-vungle @stanley-vungle We only can configure one Vungle placement for each Ad type in the IronSource Dashboard for bidding or non-biding section , it doesn't have 1:1 mapping for IS AdUnitId and Vungle placementId:
<img width="606" alt="Screen Shot 2022-10-11 at 22 44 46" src="https://user-images.githubusercontent.com/37288907/195123331-96ab8d7e-23d9-4b7f-bd0d-e9aed9bc8a28.png">

Also we cannot both enable biding and non-bidding function for a kind of ad type. For example, Banner Ad, I could only enable bidding Banner or non-bidding Banner in the IronSource Dashboard.  If I enable both of them, I cannot receive Banner ad, both John and I found this issue. 

So under these cases, the `interstitialAdapterRouter` of `ISVungleAdapter` will always correspond  to the same placementId.
Thanks a lot. :)